lib/_http_client.js

```diff
@@ -337,25 +339,47 @@ ClientRequest.prototype._implicitHeader = function _implicitHeader() {
 };
 
 ClientRequest.prototype.abort = function abort() {
-  if (!this.aborted) {
-    process.nextTick(emitAbortNT, this);
+  if (this.aborted) {
+    return;
   }
   this.aborted = true;
+  process.nextTick(emitAbortNT, this);
+  this.destroy();
+};
+
+ClientRequest.prototype.destroy = function destroy(err) {
+  if (this.destroyed) {
+    return;
+  }
+  this.destroyed = true;
 
   // If we're aborting, we don't care about any more response data.
   if (this.res) {
     this.res._dump();
```

Comment:
Question, should we leave this in `abort` and let `destroy()` be a noop of `req.writableFinished && req.res`, i.e. the response must be destroyed? This would allow us to decouple the response from the request. Which would resolve some confusion, especially when treating them as streams.

Reply:
This would allow us to `autoDestroy` the request object on `'end'`.

Reply:
@nodejs/web-server-frameworks

Reply:
What is the actually API we would like here? req and res decoupled into Writable and Readable, OR coupled into a Duplex with the `'response'` as a compat proxy?

Reply:
I guess a little out of the scope of this PR.